app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -0,0 +1,127 @@
+
+    HabitDialogListener _listener;
+
+    private EditText _habitTitleEditText;
+    private EditText _habitReasonEditText;
+    private EditText _habitDateEditText;
+    private Button _confirmNewHabit;
+    private Button _cancelNewHabit;
+    private String DATE_FORMAT = "dd-MM-yyyy";
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.add_habit_dialog_box, container, false);
+        _habitTitleEditText = view.findViewById(R.id.habit_title_edittext);
+        _habitReasonEditText = view.findViewById(R.id.habit_reason_edittext);
+        _habitDateEditText = view.findViewById(R.id.habit_date_edittext);
+        _confirmNewHabit = view.findViewById(R.id.confirmHabit);
+        _cancelNewHabit = view.findViewById(R.id.cancelHabit);
+
+        _cancelNewHabit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Cancel");
+                getDialog().dismiss();
+            }
+        });
+
+        _confirmNewHabit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.d(TAG, "Confirm");
+                String habitTitleInput = _habitTitleEditText.getText().toString();
+                String habitReasonInput = _habitReasonEditText.getText().toString();
+                String habitDateInput = _habitDateEditText.getText().toString();
+                SimpleDateFormat inputDateFormatter = new SimpleDateFormat(DATE_FORMAT);
+
+                Date habitDate = null;
+                try {
+                    habitDate = inputDateFormatter.parse(habitDateInput);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+
+                /**
+                 * When user input meets the requirements for a proper habit title, habit reason, and habit date,
+                 * have the habit be added to the habit list back in dashboard fragment.
+                 */
+                if (((habitTitleInput.length()>0)&&(habitTitleInput.length()<=20))&&
+                        ((habitReasonInput.length()>0)&&(habitReasonInput.length()<=30))&&
+                        (!(habitDateInput.equals("")))) {
+                    _listener.addNewHabit(habitTitleInput, habitReasonInput, habitDate);
+                    getDialog().dismiss();
+                } else{
+
+                    //Handle error checking for new habit name/title
+                    if (!(habitTitleInput.length()>0)){
+                        Toast.makeText(getActivity(), "Please enter a name for your habit", Toast.LENGTH_SHORT).show();
+                    } else if (!(habitTitleInput.length()<=20)){
+                        _habitTitleEditText.getText().clear();
```

Comment:
A couple of thoughts here: 

Instead of checking if it's empty or too long, we can combine these conditional statements and simply display a message that says "_Incorrect habit title entered_".

Also, I don't think we should be **clearing** the text box. The user should be able to pick up where they left off and correct their mistakes from there.